### ISSUE TITLE
add new compounds

### ISSUE THREAD

**Issue by martino79** [przemator/NSGP#1]
hi, I am a big fan of NSGP, and I am very glad that the code is free now!
It would be awesome to bring this fantastic game to a new level of complexity... 
ie add more compounds, (as in moder F1) like Super Soft, Medium, Hard.

**Comment by przemator** [comment]
Hi, unfortunately I don't have much time to invest in developing additional functionalities. If you are a developer then why don't you give it a try and I will help when I can.

**Comment by martino79** [comment]
yes, I am a c++ developer, but my experience is limited at windows env, and I usually work with Embarcadero IDE.
this nsgp how is developed? how can I get the env in witch work?

**Comment by przemator** [comment]
NSGP is written in Blitzmax, a kind of BASIC language. You first have to compile the IDE, which is possible to do in Windows, MacOS and Linux. When you have a running IDE, then you just press the button to compile NSGP.

**Comment by martino79** [comment]
hi,
I m starting to work on it...
but i need some help to start
I try to install / compile Blitmax ide from git hub, but it doesn't
compile...
like this:

***********************
***** STARTING UP *****
***********************
bin\bcc.exe
bin\bmk.exe
bin\docmods.exe
bin\FASM.EXE
bin\makedocs.exe
        5 file copiati.
lib\libunicows.a
        1 file copiati.
****************************
***** BUILDING MODULES *****
****************************
Compiling:blitz_app.c
Build Error: failed to compile
C:/Users/Martino/git_repo/blitzmax/mod/brl.mod/blitz.mod/blitz_app.cCompiling:blitz_app.c
Build Error: failed to compile
C:/Users/Martino/git_repo/blitzmax/mod/brl.mod/blitz.mod/blitz_app.c***************************
***** BUILDING MAXIDE *****
***************************
Compiling:maxide.bmx
Compile Error: Can't find interface for module 'brl.blitz'
[C:/Users/Martino/git_repo/blitzmax/mod/brl.mod/blitz.mod/blitz_classes.i;73;1]
Build Error: failed to compile
C:/Users/Martino/git_repo/blitzmax/src/maxide/maxide.bmx************************
***** BUILDING BMK *****
************************
        1 file copiati.
Compiling:bmk_config.bmx
Compile Error: Can't find interface for module 'brl.blitz'
[C:/Users/Martino/git_repo/blitzmax/mod/brl.mod/blitz.mod/blitz_classes.i;73;1]
Build Error: failed to compile
C:/Users/Martino/git_repo/blitzmax/src/bmk/bmk_config.bmx************************
***** BUILDING BCC *****
************************
Compiling:stdutil.cpp
Build Error: failed to compile
C:/Users/Martino/git_repo/blitzmax/_src/compiler/
stdutil.cpp*****************************
***** BUILDING MAKEDOCS *****
*****************************
Compiling:docnode.bmx
Compile Error: Can't find interface for module 'brl.blitz'
[C:/Users/Martino/git_repo/blitzmax/mod/brl.mod/blitz.mod/blitz_classes.i;73;1]
Build Error: failed to compile
C:/Users/Martino/git_repo/blitzmax/src/makedocs/docnode.bmx



so, I tryed also to install a pre-builded version of Blitmax IDE from site:
http://www.blitzmax.net/wordpress/download/
I have installed it and loaded NSGP.bmx: i try to compile it but it fails
for:

Building New Star GP
Compiling:New Star GP.bmx
Compile Error: Can't find interface for module 'fry.frydate'

so, I suppose that this is a "extra module" that come from git hub ide
version...

I'm quite confused.. any advice?

Thanks,
Martino




2018-04-12 19:47 GMT+02:00 Przemator <notifications@github.com>:

**Comment by przemator** [comment]
Hi, yes, I know there is an issue with compiling the module brl.blitz. This has something to do with the order of C dependencies (Blitzmax is written in C), and how MinGW handles them.

I was compiling on MacOS and after some attempt it worked, I don't know how. But I didn't manage to make it work on Windows. It's cool that you found a precompiled version of Blitzmax.

What you're missing are indeed additional modules. I put these modules in my forked version of Blitzmax:
https://github.com/przemator/blitzmax/tree/master/mod
You will need to copy whatever is missing in your version of Blitzmax. Then you will need to rebuild all modules. After that try again to compile New Star GP.bmx.

Sorry it's so messy, it was a bit frustrating to me to get it working at the beginning.

**Comment by martino79** [comment]
hi !
with some difficult, but I was able to build all: modules, and finally also
NSGP !
btw, not so difficult and the compile is quite fast.

but... I 'm concerned again.
I started a game play, first track on bahrain, and... the track is
messedup! some straight is missing, some turns are misplaced.
the game works, but the track is not playable. same problem on other tracks.

morover, i have uninstalled any previous version of the game, so i don't
undestand where it takes the trk files...

can you help someway?

tnx!
Martino







2018-04-12 22:41 GMT+02:00 Przemator <notifications@github.com>:

**Comment by przemator** [comment]
Hi, that's really cool that you managed to compile the game!

I ran into the problem of messed up track as well. I thought that I had fixed it, though.

The thing is, I am on a Mac, and here there is a difference between lowercase and uppercase. So the way tiles were loaded was messed up. Every tile has an ID: 0, 1, 2, 3... And it's important that it's loaded in an alphabetical order.

Here is the place in the code, where I thought I fixed it. Maybe debug a bit, see if the sort looks OK.
https://github.com/przemator/NSGP/blob/f06afdac9f4408b01ed9187db9fd12291f70cb6c/TTrack.bmx#L2069

If you are on Windows, he will look for the tiles in the "Media/Tiles/Track/" folder relative to the New Star GP.exe.

**Comment by przemator** [comment]
About the tracks: when you compile on Windows, the compiler will include all the files in the "Inc" folder inside the .exe file. Also, when you start the game, he will copy all the tracks to "/Documents/New Star GP".

**Comment by martino79** [comment]
Hi,
I'm still working on the project...
I want to adjust some parameters that in my opinion will make more
realistic the phisic:
nothing so much different, but:
- soft tyres have more handling and more acceleration than hard and wet.
- super soft tyres have a little more that soft
- hard tyres have no limit in top speed (why should have? I suppose the
real handicap is grip)
- wet tyres have less malus on dry condition, but degrades more fastly.
- kerFitting give lower acceleration, but no lower topspeed.
- I noticed that there is an "arcade stuff" that make AI faster if Player
is in first position: I'm not sure to like it...

But, I just ask your opinon about that points.

But there are some things that I can't understand.
- what is the parameter " randomoffspell " ?? what does it means?
- what about this code in TCar? what is the pourpose?

' Slow down more for bends when running on slicks in the wet
If tyretype <> CTYRE_WET And track.weather.wetness > 0.0
dir2wp:*((1.0+track.weather.wetness)*2)
ElseIf needpit = True And lastwayline > track.l_waylines.Count()-2
' Slow down for last few bends if tyrewear is low
dir2wp:*3
End If


thanks again!
BR
Martino


2018-04-13 21:16 GMT+02:00 Przemator <notifications@github.com>:

**Comment by przemator** [comment]
this code means that the AI car will go slower when he is driving slicks in the rain, until he reaches the pits. It's funny you mention it, one other guy was looking exactly for this line. He said that AI cars go too slow when it starts raining.

Btw, if you download Github Desktop, then you can track your changes, you can also create a pull request once you implement a working change. Then I can review it and merge it into the master version. This way we can see the version history and keep everything in one place.

**Comment by przemator** [comment]
Regarding your ideas:
You're definitely right about the speed limit. The difference between the tyres should not be about acceleration or top speed, but about grip in the corners. When it comes to KERS, I haven't been following F1 for some time, but I thought that both cars with and without KERS weight the same. The actual car weighs much less than the allowed minimum limit, and the teams put ballast to make it heavier. If you have KERS, this means you cannot put ballast as freely, so the handling of the car is worse. So I don't think having KERS should make aceeleration or top speed worse.
